### ISSUE TITLE
[szip] CMake Error: try_run() invoked in cross-compiling mode

### ISSUE THREAD

**Issue by MistEO** [microsoft/vcpkg#34102]
### Operating system

Linux and macOS Cross-compile arm64 binaries using x64 containers

### Steps to reproduce the behavior

see https://github.com/MaaAssistantArknights/MaaDeps/actions/runs/6335612832/job/17207088613 and https://github.com/MaaAssistantArknights/MaaDeps/actions/runs/6335612832/job/17207089437

Linux and macOS Cross-compile arm64 binaries using x64 containers

### Failure logs

```log
-- Looking for 13 include files unistd.h, ..., fcntl.h
-- Looking for 13 include files unistd.h, ..., fcntl.h - found
-- Looking for 14 include files unistd.h, ..., inttypes.h
-- Looking for 14 include files unistd.h, ..., inttypes.h - found
CMake Error: try_run() invoked in cross-compiling mode, please set the following cache variables appropriately:
   HAVE_DEFAULT_SOURCE_RUN (advanced)
   HAVE_DEFAULT_SOURCE_RUN__TRYRUN_OUTPUT (advanced)
For details see /home/runner/work/MaaDeps/MaaDeps/vcpkg/buildtrees/szip/maa-arm64-linux-rel/TryRunResults.cmake
CMake Error: try_run() invoked in cross-compiling mode, please set the following cache variables appropriately:
   TEST_LFS_WORKS_RUN (advanced)
   TEST_LFS_WORKS_RUN__TRYRUN_OUTPUT (advanced)
For details see /home/runner/work/MaaDeps/MaaDeps/vcpkg/buildtrees/szip/maa-arm64-linux-rel/TryRunResults.cmake
-- Performing TEST_LFS_WORKS... no
-- Performing Other Test HAVE_OFF64_T - Failed
```

Full log: 
[config-maa-arm64-linux-out.log](https://github.com/microsoft/vcpkg/files/12750383/config-maa-arm64-linux-out.log)
[buildtrees.zip](https://github.com/microsoft/vcpkg/files/12750405/logs-arm64-linux.2.zip)

**Comment by dg0yt** [comment]
This is the standard szip cmake behaviour, not a port bug. They rely on try-run or user input for certain configuration elements.

If you really want to try it out, you can pass the suggested options from your custom triplet file, e.g.
~~~
if(PORT STREQUAL "szip")
    list(APPEND VCPKG_CMAKE_CONFIGURE_OPTIONS
        -DTEST_LFS_WORKS_RUN=TRUE
        -DTEST_LFS_WORKS_RUN__TRYRUN_OUTPUT=
        -DHAVE_DEFAULT_SOURCE_RUN=0
        -DHAVE_DEFAULT_SOURCE_RUN__TRYRUN_OUTPUT=
endif()
~~~
These are the value the portfile applies for Android/iOS/macOS cross targets.
i.e. "macOS Cross-compile arm64 binaries using x64" shoudl already work.

What do you want to use it for? hdf5 has the same problem.

**Comment by neko-para** [comment]
> This is the standard szip cmake behaviour, not a port bug. They rely on try-run or user input for certain configuration elements.
> 
> If you really want to try it out, you can pass the suggested options from your custom triplet file, e.g.
> 
> ```cmake
> if(PORT STREQUAL "szip")
>     list(APPEND VCPKG_CMAKE_CONFIGURE_OPTIONS
>         -DTEST_LFS_WORKS_RUN=TRUE
>         -DTEST_LFS_WORKS_RUN__TRYRUN_OUTPUT=
>         -DHAVE_DEFAULT_SOURCE_RUN=0
>         -DHAVE_DEFAULT_SOURCE_RUN__TRYRUN_OUTPUT=
>     )
> endif()
> ```
> 
> These are the value the portfile applies for Android/iOS/macOS cross targets. i.e. "macOS Cross-compile arm64 binaries using x64" shoudl already work.
> 
> What do you want to use it for? hdf5 has the same problem.

But, according to the action provided above, szip DOES configured on Windows while targetting arm64. Is there any difference between win and other os?

**Comment by neko-para** [comment]
![image](https://github.com/microsoft/vcpkg/assets/26452623/affac7ee-24ec-4510-a0a8-8fb03339abb3)
@dg0yt 
I've found this in portfile.cmake of szip. It seems that we've already provided a preset configuration for mac target, just like you've mentioned above. But we still failed building on mac. Could you please give us some suggestions?

**Comment by dg0yt** [comment]
> I've found this in portfile.cmake of szip.

That's what I said: "These are the value the portfile applies for Android/iOS/macOS cross targets."

> Szip is built on mac

So my claim is confirmed: "macOS Cross-compile arm64 binaries using x64" shoudl already work."

> but hfd5 don't.

And that's why I wrote early: "What do you want to use it for? hdf5 has the same problem."
It is a know issue for years. hdf5 1.15 will come with some improvements, but ATM there are still some `try_run` tests left.

**Comment by neko-para** [comment]
After some of tries, I've managed to pass the configuring step. However I'm stuck in the build step, as it tries to run some arm binary. I decided to setup qemu and set the CMAKE_CROSSCOMPILING_EMULATOR. But I cannot get the libc6:arm64 installed, as `apt update` keeps failed. Is there any option to let hdf5 static link those detecting programs?

---

```yml
      - uses: ryankurte/action-apt@v0.3.0
        if: runner.os == 'Linux' && matrix.arch == 'arm64'
        with:
          arch: arm64
          packages: 'libc6:arm64'
```
